[Processing] Load only providers from processing module when initialising Processing class

radosuav opened the pull request:
Currently, Processing.initialize() loads providers coming from additional plugins together with the basic ones. This might not be always desirable and also it forces the additional providers to have init function with no parameters. 

This affects 2.16 branch as well.

radosuav committed:
[Processing] Load only providers from processing module when initialising Processing class

radosuav commented:
@volaya @alexbruy - would it be possible for you to have a look at this PR this week? Would be nice if it was included in 2.16.1. Thanks!

volaya commented:
I don't see the issue with the initialization, but if those additional providers are not initialized, how can they later be added to processing? they won't appear on the settings, and won't show as result of the search box in the toolbox. Can you provide more info of why this change is needed?

Thanks!

radosuav commented:
The additional providers are initialized and added to the Processing by their plugins (e.g. see https://github.com/DHI-GRAS/processing_workflow/blob/GW-A_dev/ProcessingWorkflowPlugin.py which is based on https://github.com/qgis/QGIS/blob/master/python/plugins/processing/algs/exampleprovider/ProcessingExampleProviderPlugin.py).

This PR restores the situation from before https://github.com/qgis/QGIS/commit/e6b76c4d7bf2f07a1b5817261d41281f7ef88dc4#diff-681e98194d7e5da32e7231f6f98ac552L142, i.e. only the providers which come with Processing are initialized and added in Processing initialize(). As far as I understand the aim of that code change was to clean up the code and not to modify the behavior but please let me know if I have missed something.

The problem arises when the additional providers have init() function with parameters. As an example WorkflowProvider is initialized with iface parameter (https://github.com/DHI-GRAS/processing_workflow/blob/GW-A_dev/WorkflowProvider.py#L46). I know that I could get iface in another way but other providers might require other parameters during initialization.

alexbruy commented:
With ongoing efforts on moving Processing into core this is not relevant anymore
